src/main/java/ch/njol/skript/expressions/ExprBookPages.java

```diff
@@ -19,99 +19,189 @@
 
 import ch.njol.skript.Skript;
-import ch.njol.skript.aliases.Aliases;
 import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.event.Event;
-import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.BookMeta;
 import org.eclipse.jdt.annotation.Nullable;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 @Name("Book Pages")
 @Description("The pages of a book.")
-@Examples({"on book sign:",
-			"\tmessage \"Book Pages: %pages of event-item%\"",
-			"\tmessage \"Book Page 1: %page 1 of event-item%\""})
-@Since("2.2-dev31")
+@Examples({
+	"on book sign:",
+	"\tmessage \"Book Pages: %pages of event-item%\"",
+	"\tmessage \"Book Page 1: %page 1 of event-item%\"",
+	"set page 1 of player's held item to \"Book writing\""
+})
+@Since("2.2-dev31, INSERT VERSION (changers)")
 public class ExprBookPages extends SimpleExpression<String> {
-	
+
 	static {
 		Skript.registerExpression(ExprBookPages.class, String.class, ExpressionType.PROPERTY,
-				"[all] [the] [book] (pages|content) of %itemtypes%",
-				"%itemtypes%'s [book] (pages|content)",
-				"[book] page %number% of %itemtypes%", 
-				"%itemtypes%'s [book] page %number%");
+			"[(all [[of] the]|the)] [book] (pages|content) of %itemtypes%",
+			"%itemtypes%'[s] [book] (pages|content)",
+			"[book] page %number% of %itemtypes%",
+			"%itemtypes%'[s] [book] page %number%");
 	}
-	
-	private static final ItemType bookItem = Aliases.javaItemType("book with text");
-	
-	@SuppressWarnings("null")
-	private Expression<ItemType> book;
+
+	private Expression<ItemType> items;
 	@Nullable
 	private Expression<Number> page;
-	
-	@SuppressWarnings("null")
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 0 || matchedPattern == 1) {
+			items = (Expression<ItemType>) exprs[0];
+		} else {
+			if (matchedPattern == 2) {
+				page = (Expression<Number>) exprs[0];
+				items = (Expression<ItemType>) exprs[1];
+			} else {
+				items = (Expression<ItemType>) exprs[0];
+				page = (Expression<Number>) exprs[1];
+			}
+		}
+		return true;
+	}
+
+	@Override
 	@Nullable
+	@SuppressWarnings("null")
+	protected String[] get(Event event) {
+		List<String> pages = new ArrayList<>();
+		for (ItemType itemType : items.getArray(event)) {
+			if (itemType == null || !(itemType.getItemMeta() instanceof BookMeta))
```

Comment:
```suggestion
			if (!(itemType.getItemMeta() instanceof BookMeta))
```
null check shouldn't be necessary

Reply:
Was there a reason this was marked as resolved, I don't remember :(